src/NBitcoin/ProvenBlockHeader.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Linq;
+
+namespace NBitcoin
+{
+    /// <summary>
+    /// Extension to an existing <see cref="BlockHeader"/> which is used in PoS to prevent attacker from constructing
+    /// a fake chain of headers that has more work than the valid chain and attacking a node.
+    ///
+    /// Proven header prevents such an attack by including additional information that can be validated and confirmed whether
+    /// the header is fake or real.
+    ///
+    /// Additional information included into proven header:
```

Comment:
From here to the end, it should probably be the remarks section

Reply:
Done